salt/states/boto_ec2.py

```diff
@@ -278,6 +296,90 @@ def eni_present(
+            else:
+                eip_alloc = __salt__['boto_ec2.allocate_eip_address'](domain=None,
+                                                                      region=region,
+                                                                      key=key,
+                                                                      keyid=keyid,
+                                                                      profile=profile)
+                if eip_alloc:
+                    _ret = __salt__['boto_ec2.associate_eip_address'](instance_id=None,
+                                                                      instance_name=None,
+                                                                      public_ip=None,
+                                                                      allocation_id=eip_alloc['allocation_id'],
+                                                                      network_interface_id=r['result']['id'],
+                                                                      private_ip_address=None,
+                                                                      allow_reassociation=False,
+                                                                      region=region,
+                                                                      key=key,
+                                                                      keyid=keyid,
+                                                                      profile=profile)
+                    if not _ret:
+                        _ret = __salt__['boto_ec2.release_eip_address'](public_ip=None,
+                                                                        allocation_id=eip_alloc['allocation_id'],
+                                                                        region=region,
+                                                                        key=key,
+                                                                        keyid=keyid,
+                                                                        profile=profile)
+                        ret['result'] = False
+                        msg = 'Failed to assocaite the allocated EIP address with the ENI.  The EIP {0}'.format('was successfully released.' if _ret else 'was NOT RELEASED.')
+                        ret['comment'] = ' '.join([ret['comment'], msg])
+                        return ret
+                else:
+                    ret['result'] = False
+                    ret['comment'] = ' '.join([ret['comment'], 'Failed to allocate an EIP address'])
+                    return ret
+        else:
+            ret['comment'] = ' '.join([ret['comment'], 'An EIP is already allocated/assocaited to the ENI'])
+    if arecords:
+        for arecord in arecords:
+            if 'name' not in arecord:
+                msg = 'The arecord must contain a "name" property.'
+                raise SaltInvocationError(msg)
+            if __opts__['test']:
```

Comment:
I think it's actually ok to not have test mode here. The state module you're calling has test mode and will handle this correctly.